src/libs/antares/study/include/antares/study/parts/hydro/finalLevelValidator.h

```diff
@@ -0,0 +1,75 @@
+/*
+** Copyright 2007-2023 RTE
+** Authors: Antares_Simulator Team
+**
+** This file is part of Antares_Simulator.
+**
+** Antares_Simulator is free software: you can redistribute it and/or modify
+** it under the terms of the GNU General Public License as published by
+** the Free Software Foundation, either version 3 of the License, or
+** (at your option) any later version.
+**
+** There are special exceptions to the terms and conditions of the
+** license as they are applied to this software. View the full text of
+** the exceptions in file COPYING.txt in the directory of this software
+** distribution
+**
+** Antares_Simulator is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+** GNU General Public License for more details.
+**
+** You should have received a copy of the GNU General Public License
+** along with Antares_Simulator. If not, see <http://www.gnu.org/licenses/>.
+**
+** SPDX-License-Identifier: licenceRef-GPL3_WITH_RTE-Exceptions
+*/
+#pragma once
+
+#include <vector>
+#include <antares/array/matrix.h>
+
+namespace Antares::Data
+{
+class PartHydro;
+
+
+class FinalLevelValidator
```

Comment:
This source file replaces the old **finallevelinflowsmodifyer.h** and this class replaces old class **FinalLevelInflowsModifier**.

Actually, the computation and handling of **deltas** (final level - initial level) was moved elsewhere, and the class focuses on a **particular year**, which was not the case previously.

it's meant to be more simple and clear : 
- its name is more understandable
- it no more have an **initialize(...)** function but only a constructor with more arguments
- data members are a bit less numerous and more simple (no more **Matrix**)
- member functions' names are meant to be more meaningful
-  the class does a bit less things (by the way, the associated **.cpp** is smaller).